### ISSUE TITLE
TextPath and TSpan need platform-specific mappiings

### ISSUE THREAD

**Issue by joanmarie** [w3c/svg-aam#2]
`TextPath` and `TSpan` should be mapped as `ATK_ROLE_STATIC`. The current mapping to `group` role means they will wind up exposed as `ATK_ROLE_PANEL`.

This was originally filed against ARIA (i.e. before the Great Repo Split of 2017): https://github.com/w3c/aria/issues/133

**Comment by joanmarie** [comment]
Regarding the needs implementer input label:

As an ATK implementor (I did the SVG AAM implementation in WebKit a couple of years ago) and as the maintainer of the GNU/Linux screen reader, I trust you don't need further input from me.

If you need input for other platforms, perhaps pinging them to comment on this issue would be helpful?

**Comment by AmeliaBR** [comment]
@joanmarie Yes, the input required is for the other platforms. But at the time I posted it, I was mostly focused on cleaning things up quickly for the republished WD.

Maybe @melanierichards and @cookiecrook could take a look (or suggest someone else who could)?

See the [spec](https://w3c.github.io/svg-aam/#textpath-tspan-mappings-issue) for more details.

The question is basically: what's the appropriate platform API role for a formatted text span that needs to be exposed as an accessible object with name/description?

**Comment by joanmarie** [comment]
@asurkov could you please provide input w.r.t. IA2? Thanks!

**Comment by melanierichards** [comment]
Thanks for the ping @AmeliaBR and @joanmarie, `Control Type: Group` (how we map the `group` role) would still be the most sensible mapping for `textPath` and `tspan` in UIA.

**Comment by asurkov** [comment]
@joanmarie, IA2_ROLE_TEXT_FRAME looks appropriate for tpath element. However, I'm not sure whether tspan element should have accessible object, because it looks quite close to HTML:span element, used for text styling, which should be expressed by text attributes. The latter is also valid for ATK case I would say.

**Comment by joanmarie** [comment]
@asurkov The mapping tables describe what should happen if the object is included in the accessibility tree. So if the object is excluded from the tree, it wouldn't get mapped.

As with span, if it were used for styling and, say, made focusable, that span/tspan would need to be in the accessibility tree of course. And it would need a role which makes sense. :)

**Comment by asurkov** [comment]
oh, if there's no requirement to make it having an accessible object, then it's fine. So if it has to have an accessible object for some reason, for example, it is focusable, then IA2_ROLE_TEXT_FRAME should work.
